benchmarks/bench_micro.c

```diff
@@ -0,0 +1,333 @@
+#include <stdio.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+#include <pthread.h>
+
+#include <storage/cuckoo/item.h>
+#include <storage/cuckoo/cuckoo.h>
+#include <time/cc_timer.h>
+#include <cc_debug.h>
+#include <cc_mm.h>
+#include <cc_array.h>
+
+static __thread unsigned int rseed = 1234; /* XXX: make this an option */
+
+#define RRAND(min, max) (rand_r(&(rseed)) % ((max) - (min) + 1) + (min))
+
+#define SWAP(a, b) do {\
+    __typeof__(a) _tmp = (a);\
+    (a) = (b);\
+    (b) = _tmp;\
+} while (0)
+
+#define BENCHMARK_OPTION(ACTION)\
+    ACTION(entry_min_size,  OPTION_TYPE_UINT, 64,    "Min size of cache entry")\
+    ACTION(entry_max_size,  OPTION_TYPE_UINT, 64,    "Max size of cache entry")\
+    ACTION(nentries,        OPTION_TYPE_UINT, 1000,  "Max total number of cache entries" )\
+    ACTION(nops,            OPTION_TYPE_UINT, 100000,"Total number of operations")\
+    ACTION(pct_get,         OPTION_TYPE_UINT, 80,    "% of gets")\
+    ACTION(pct_put,         OPTION_TYPE_UINT, 10,    "% of puts")\
+    ACTION(pct_rem,         OPTION_TYPE_UINT, 10,     "% of removes")
+
+#define O(b, opt) option_uint(&(b->options.benchmark.opt))
+
+typedef size_t benchmark_key_t;
+
+struct benchmark_entry {
+    char *key;
+    benchmark_key_t key_size;
+
+    char *value;
+    size_t value_size;
+};
+
+struct benchmark_specific {
+    BENCHMARK_OPTION(OPTION_DECLARE);
+};
+
+struct benchmark_options {
+    struct benchmark_specific benchmark;
+    cuckoo_options_st cuckoo;
+};
+
+struct benchmark {
+    FILE *config;
+    struct benchmark_entry *entries;
+    struct benchmark_options options;
+};
+
+static int
```

Comment:
nit: for the sake of consistency with the rest of the codebase, consider using rstatus_i and statuses defined in [cc_define.h](https://github.com/twitter/ccommon/blob/master/include/cc_define.h)